### PR TITLE
Fix google docs lists

### PR DESCRIPTION
**Pull request by kevindew**
Trello: https://trello.com/c/P5KBetGh/754-resolve-issues-identified-from-copy-and-paste-testing

See commits for details

### CODE REVIEW

File: src/to-govspeak.js [hunk 1]
```diff
@@ -114,6 +114,54 @@ service.addRule('cleanUpNestedLinks', {
...
+
+    let prefix = options.bulletListMarker + '   '
+    const parent = node.parentNode
+    if (parent.nodeName.toLowerCase() === 'ol') {
+      const start = parent.getAttribute('start')
+      const listItems = Array.prototype.filter.call(
+        parent.children, (element) => element.nodeName.toLowerCase() === 'li'
+      )
+      const index = Array.prototype.indexOf.call(listItems, node)
+      prefix = (start ? Number(start) + index : index + 1) + '.  '
```
Comment (benthorner) [inline src/to-govspeak.js]: I'm afraid the interaction between 'start' and 'index' is confusing me, but it does look like there's more than one case to test here. Would it be possible to break this up in some way, to make it more transparent?

Comment (kevindew) [reply src/to-govspeak.js]: I don't really want to change the code here as I want to maintain it being as close to the ported code as possible, but happy to add a test for the start condition. 2 mins.

Comment (kevindew) [reply src/to-govspeak.js]: Actually just tried out the start code and it does nothing for us when the markdown is converted back to govspeak so will just remove it.

Example:

HTML
```html
<ol start=5>
  <li>item 1</li>
  <li>item 1</li>
</ol>
```

Converts to govspeak/md:

```
5.  item
6.  item
```

Then from govspeak to HTML:

```html
<div class="gem-c-govspeak govuk-govspeak ">
  <ol>
     <li>item</li>
     <li>item</li>
   </ol>
</div>
```

Comment (kevindew) [reply src/to-govspeak.js]: I've updated now

**Comment by kevindew** [comment]
> Thanks for clarifying - I think I understand it enough to see that it works now :-). Just a few questions, which I don't think are blocking:
> 
> * It looks like we depend on the indentation of the input HTML in order to do the actual nested. Is that right?

Nope, that's not the case. An input of the below still passes tests:

```
  const html = `
    <ol>
      <li>Parent</li>
      <ol>
        <li>Child 1</li>
    <ol>
      <li>Grand child 1</li>
      <li>Grand child 2</li>
    </ol>
        <li>Child 2</li>
        <li>Child 3</li>
      </ol>
      <li>Parent sibling</li>
    </ol>
  `
```

> * I notice the conversion actually puts 3 spaces after the number/bullet marker, which is a bit weird and different to what our toolbar generates. But then I see it's also what the origin code does, so I'm guessing we're compromising here?

That's a good point, we were just dealing with that before but since I've hijacked the listItem markup I can update that to match what we have (hopefully) - I'll give that a whirl.

**Comment by kevindew** [comment]
@benthorner I've set the spacing up now.

**Comment by benthorner** [comment]
👍

**Pull request merged**